osier/models/dispatch.py

```diff
@@ -78,30 +107,144 @@ class DispatchModel():
     model : :class:`pyomo.environ.ConcreteModel`
         The :mod:`pyomo` model class that converts python code into a
         set of linear equations.
-    upperbound : float
+    objective : float
+        The result of the model's objective function. Only instantiated
+        after :meth:`DispatchModel.solve()` is called.
+    n_timesteps : int
+        The number of timesteps in the model.
+    upper_bound : float
         The upper bound for all decision variables. Chosen to be equal
         to the maximum capacity of all technologies in :attr:`tech_set`.
+    storage_upper_bound : float
+        The upper bound for storage decision variables.
+    penalty : float
+        The penalty applied to the objective function to eliminate
+        simultaneous charging and discharging. Users may need to tune
+        this parameter. Default is 1.0.
+
+    model_initialized : boolean
+        Indicates whether :attr:`DispatchModel.model` has been populated
+        with equations yet. This is set to ``True`` after
+        :meth:`DispatchModel._write_model_equations()` has been called.
+    indices : List of tuples
+        The list of tuples representing the product of the
+        :attr:`tech_set` and :attr:`time_set` attributes.
+    tech_set : list of str
+        A list of the unique technology names in the simulation.
+    capacity_dict : dict
+        A dictionary of {name : capacity} pairs.
+    efficiency_dict : dict
+        A dictionary of {name : efficiency} pairs.
+    time_set : iterable
+        The result of ``range(self.n_timesteps)``.
+    cost_params : list
+        The set of cost parameters for each technology. Corresponds to
+        a list of values. Size is equal to the product of the number of
+        timesteps and the number of technologies in the model.
+    ramp_up_params : list
+        The set of ramp_up parameters. Only initialized if there is a
+        ramping technology.
+    ramp_down_params : list
+        The set of ramp_down parameters. Only initialized if there is a
+        ramping technology.
```

Comment:
They are list of floats, but again this is not something that users can modify.

Reply:
So if a user puts in a list of integers, it will get converted to a list of floats? Is there any possibility that the user could give these as a list of strings? Just want to cover any bases for documentation to reduce user errors.

Reply:
A user cannot set or change this attribute except by modifying the technology objects in 'technology_list'. The technology objects may accept a string for this value but it is internally converted to a unyt quantity and subsequently to a float. This has an @property decorator but no corresponding @setter method, so users can't change it once the class is initialized.